core/authz_test.go

```diff
@@ -0,0 +1,161 @@
+	err = os.Mkdir(raftDir, 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(raftDir)
+
+	raftDB, err := raft.Start("", raftDir, "", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	api := &API{
+		mux:          http.NewServeMux(),
+		raftDB:       raftDB,
+		accessTokens: accessTokens,
+	}
+	api.buildHandler()
+	server := httptest.NewServer(api)
+	defer server.Close()
+
+	testPolicies := []string{
+		"client-readwrite",
+		"client-readonly",
+		"network",
+		"monitoring",
+		"internal",
+		"public",
+	}
+	tokens := make(map[string]*accesstoken.Token)
+	for i := 0; i < len(testPolicies); i++ {
+		token, err := accessTokens.Create(ctx, fmt.Sprintf("token%d", i), "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		tokens[policies[i]] = token
+		err = api.createGrant(ctx, apiGrant{
+			GuardType: "access_token",
+			GuardData: map[string]interface{}{"id": token.ID},
+			Policy:    policies[i],
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	pathTokenMap := map[string]map[string]bool{
+		"/create-account": map[string]bool{
+			"client-readwrite": true,
+			"client-readonly":  false,
+			"network":          false,
+			"monitoring":       false,
+			"internal":         false,
+			"public":           false,
+		},
+		"/list-accounts": map[string]bool{
+			"client-readwrite": true,
+			"client-readonly":  true,
+			"network":          false,
+			"monitoring":       false,
+			"internal":         false,
+			"public":           false,
+		},
+		"/reset": map[string]bool{
+			"client-readwrite": true,
+			"client-readonly":  false,
+			"network":          false,
+			"monitoring":       false,
+			"internal":         true,
+			"public":           false,
+		},
+		networkRPCPrefix + "get-block": map[string]bool{
+			"client-readwrite": false,
+			"client-readonly":  false,
+			"network":          true,
+			"monitoring":       false,
+			"internal":         false,
+			"public":           false,
+		},
+		"/info": map[string]bool{
+			"client-readwrite": true,
+			"client-readonly":  true,
+			"network":          true,
+			"monitoring":       true,
+			"internal":         false,
+			"public":           false,
+		},
+		"/debug/pprof/symbol": map[string]bool{
+			"client-readwrite": true,
+			"client-readonly":  true,
+			"network":          false,
+			"monitoring":       true,
+			"internal":         false,
+			"public":           false,
+		},
+		"/docs": map[string]bool{ // public is open to all
+			"client-readwrite": true,
+			"client-readonly":  true,
+			"network":          true,
+			"monitoring":       true,
+			"internal":         true,
+			"public":           true,
+		},
+	}
+
+	for path, policyMap := range pathTokenMap {
+		for policy, want := range policyMap {
+			got := tryEndpoint(t, server.URL, path, tokens[policy])
+			if got != want {
+				t.Errorf("auth(%s, %s) = %t want %t", path, policy, got, want)
+			}
+		}
+	}
+}
+
+func tryEndpoint(t testing.TB, baseURL, path string, token *accesstoken.Token) bool {
```

Comment:
Nit: can we call it tryRPC instead? (I am still fighting my (losing) battle against calling these things "endpoints".)